mscore/scoretab.cpp

```diff
@@ -42,6 +42,26 @@
 
 namespace Ms {
 
+//---------------------------------------------------------
+//   MsTabBar
+//    Implemented the ability of closing tab by middle click
+//---------------------------------------------------------
+
+MsTabBar::MsTabBar(QWidget* parent)
+   : QTabBar(parent)
+      {}
+
+//---------------------------------------------------------
+//   MsTabBar::mouseReleaseEvent
+//---------------------------------------------------------
+
+void MsTabBar::mouseReleaseEvent(QMouseEvent* event)
+      {
+      QTabBar::mouseReleaseEvent(event);
+      if (event->button() == Qt::MiddleButton)
+            emit tabCloseRequested(tabAt(event->pos()));
```

Comment:
If user clicks middle button somewhere else, moves mouse to the tab bar and releases the mouse button this code will still make this tab close. It doesn't seem to be desirable in this case.

Reply:
This would be hard to fix. Plus, this only happen when the mouse is pressed _within the tab bar zone_. If you want to abort closing a tab using middle click, you'd most likely move the mouse outside the tab bar rather than moving to a different tab. So I think this is not a big deal.

Reply:
Why not override `mousePressEvent()` and set an internal flag, then check inside `mouseReleaseEvent()` that the flag is set before closing the tab?

Reply:
Coincidentally, I'm working on it using the exact method ;-)